packages/transaction-controller/src/TransactionController.ts

```diff
@@ -437,6 +441,9 @@ export class TransactionController extends BaseController<
       securityAlertResponse,
       // Add actionId to txMeta to check if same actionId is seen again
       actionId,
+      type:
```

Comment:
Should we calculate this in a separate variable above so all this logic isn't inline?

Reply:
Done, I've broken down the logic.